src/operations/common/RepoId.ts

```diff
@@ -82,7 +82,7 @@ export class RemoteRepoRefSupport implements RemoteRepoRef {
     }
 
     public cloneUrl(creds: ProjectOperationCredentials) {
-        return `https://${creds.token}@${this.remoteBase}/${this.pathComponent}.git`;
+        return `https://${creds.token}:x-oauth-basic@${this.remoteBase}/${this.pathComponent}.git`;
```

Comment:
this is wonderful. Does it mean if I pass garbage as a token, it won't ask me for that user's password?

Reply:
Correct. That's the issue I ran into and was seeing that test fail. Hence the fix. Pulling that thread let to a couple other fixes.